src/components/Header/Header.js

```diff
@@ -1,25 +1,49 @@
 import React, { Component } from 'react';
 import { Link } from 'react-router-dom';
+import Button from '@material-ui/core/Button';
+import Menu from '@material-ui/core/Menu';
+import MenuItem from '@material-ui/core/MenuItem';
 import style from './style.scss';
 
 export default class Header extends Component {
   state = {
     isAdmin: false,
     logIn: false,
-    userName: ''
+    userName: '',
+    anchorEl: null
+  };
+
+  handleClick = event => {
+    this.setState({ anchorEl: event.currentTarget });
+  };
+
+  handleClose = () => {
+    this.setState({ anchorEl: null });
+  };
+
+  logOut = () => {
+    this.setState({ anchorEl: null });
+    localStorage.removeItem('token');
+    localStorage.removeItem('userName');
   };
 
   componentWillMount = () => {
     const userName = localStorage.getItem('userName');
-    userName ? this.setState({logIn: true, userName}) : this.setState({logIn: false})
-  }
+    const token = localStorage.getItem('token');
+    token
+      ? this.setState({ logIn: true, userName })
+      : this.setState({ logIn: false });
+  };
 
   componentWillReceiveProps = () => {
```

Comment:
componentWillMount and componentWillReceiveProps are both deprecated
https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html